src/main/java/com/linkedin/kmf/services/Service.java

```diff
@@ -9,15 +9,39 @@
  */
 package com.linkedin.kmf.services;
 
+/**
+ * Services are components of a monitoring application that are expected to be running continuously in order to perform
+ * monitoring.
+ */
 public interface Service {
 
   static final String JMX_PREFIX = "kmf.services";
 
+  /**
+   * The start logic must only execute once.  If an error occurs then the implementer of this class must assume that
+   * stop() will be called to clean up.  This method must be thread safe and must assume that stop() may be called
+   * concurrently. This can happen if the monitoring application's life cycle is being managed by a container.  Start
```

Comment:
What does container mean in the open source world? Do you mean the multi-product container?